packages/sources/svelte/src/components/selection-controls/VtmnCheckbox/VtmnCheckbox.svelte

```diff
@@ -0,0 +1,53 @@
+<script>
+  import { cn } from '../../../utils/classnames';
+
+  /**
+   * ID of the checkbox and associated label
+   * @type {string}
+   */
+  export let id;
+
+  /**
+   * Whether the checkbox is indeterminate
+   * @type {boolean}
+   */
+  export let indeterminate = false;
+
+  /**
+   * Whether the checkbox is disabled
+   * @type {boolean}
+   */
+  export let checked = false;
+
+  /**
+   * Whether the checkbox is disabled
+   * @type {boolean}
+   */
+  export let disabled = false;
+
+  let className;
+  /**
+   * @type {string} Custom classes to apply to the component.
+   */
+  export { className as class };
+
+  $: componentClass = cn(
+    'vtmn-checkbox',
+    indeterminate && checked && 'vtmn-checkbox--indeterminate',
```

Comment:
The component don't need to be checked to add the indeterminate class
`indeterminate && 'vtmn-checkbox--indeterminate'`